data/src/main/java/com/microsoft/azure/kusto/data/exceptions/DataServiceException.java

```diff
@@ -11,4 +11,8 @@ public DataServiceException(String ingestionSource, String message, boolean isPe
     public DataServiceException(String ingestionSource, String message, Exception exception, boolean isPermanent) {
         super(ingestionSource, message, exception, isPermanent);
     }
+
+    public boolean is404Error() {
+        return getCause() != null && getCause() instanceof DataWebException && ((DataWebException) getCause()).getHttpResponse().getStatusLine().getStatusCode() == 404;
```

Comment:
Can it be a 404 error and not be of type DataWebException?

Reply:
It can not. Look at `createExceptionFromResponse`
DataWebException is our type, and we create it whenever we get an http response which is not 200